src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs

```diff
@@ -382,6 +632,77 @@ public void AutomaticLineFormat()
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void NotAfterExisitingSemicolon()
+        {
+            Test(@"class TestClass
+{
+    private int i;
+    $$
+}", @"class TestClass
+{
+    private int i;$$
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void NotAfterCloseBraceInMethod()
+        {
+            Test(@"class TestClass
+{
+    void Test() { }
+    $$
+}", @"class TestClass
+{
+    void Test() { }$$
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void NotAfterCloseBraceInStatement()
+        {
+            Test(@"class TestClass
+{
+    void Test()
+    {
+        if (true) { }
+        $$
+    }
+}", @"class TestClass
+{
+    void Test()
+    {
+        if (true) { }$$
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void NotAfterAutoPropertyAccessor()
```

Comment:
Didn't change anything here, just asserting existing behavior and also protecting against regressions since i relaxed the code a bit by removing a check for close parenthesis.

The code doesn't explicitly block against completing here though. But i guess it won't work as such because we only complete in cases where semicolon is the last token on the line.. In case of accessors, semicolon is not the last token on line, the close paren is.

If this would be worthwhile, I can file a workitem and address it separately :smile: